packages/integrations/vue/src/index.ts

```diff
@@ -80,6 +83,20 @@ export const setup = (app) => {
 				return `export const setup = () => {};`;
 			}
 		},
+		// Ensure that Vue components reference appEntrypoint directly
+		// This allows Astro to assosciate global styles imported in this file
+		// with the pages they should be injected to
+		transform(code, id) {
+			if (!appEntrypoint) return;
+			if (id.endsWith('.vue')) {
+				const s = new MagicString(code);
+				s.prepend(`import "${appEntrypoint}";\n`);
+				return {
+					code: s.toString(),
+					map: s.generateMap()
+				}
+			}
+		},
```

Comment:
Does this mean if there are 10 components, a css file will be imported 10 times?

Reply:
Imported 10 times, but not included in the parent page 10 times.

Reply:
Is that some vite magic that deduplicates those? Genuinely curious

Reply:
Yes, it's the same as if you imported A.astro and B.astro into your page, while A in turn also imported B.